### PR TITLE
[SYCL] populate PI (and direct SYCL RT to PI)

### PR DESCRIPTION
**Pull request by smaslov-intel**
This incremental change on top of PI infrastructure being added in #222. It turns all of SYCL RT OpenCL calls into PI. For the time being PI implementation is still directly the OpenCL one (in pi_opencl.*), and the next step will be to run OpenCL through the PI.

### CODE REVIEW

**Comment by smaslov-intel** [comment]
Review the commit 
f83758d: https://github.com/intel/llvm/pull/259/commits/f83758df6dbc7784f2a0f9614c4eb3c251ab5e7c

File: sycl/include/CL/sycl/detail/pi.h [hunk 1]
```diff
@@ -73,10 +74,35 @@ typedef enum {
...
+} _pi_queue_info;
+
+// NOTE: this is made 64-bit to match the size of cl_mem_flags to
+// make the translation to OpenCL transparent.
+// TODO: populate
+//
+typedef pi_uint64 pi_mem_flags;
+// Access
+const pi_uint64 PI_MEM_FLAGS_ACCESS_RW     = CL_MEM_READ_WRITE;
+  // Host pointer
```
[the remark this answers is not in the document]
Comment (smaslov-intel) [reply sycl/include/CL/sycl/detail/pi.h]: OK

File: sycl/include/CL/sycl/detail/pi_opencl.hpp [hunk 2]
```diff
@@ -1,14 +1,18 @@
...
 // SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
 //
 //===----------------------------------------------------------------------===//
+//
+// This is doing temporary redirection of PI to OpenCL at compile-time.
+// TODO: when PI is ready get rid of this file.
+//
 #pragma once
 
+#include <CL/sycl/detail/pi.h>
```
[the remark this answers is not in the document]
Comment (smaslov-intel) [reply sycl/include/CL/sycl/detail/pi_opencl.hpp]: OK, should be possible in this case

File: sycl/include/CL/sycl/detail/queue_impl.hpp [hunk 3]
```diff
@@ -41,27 +41,31 @@ class queue_impl {
   queue_impl(cl_command_queue CLQueue, const context &SyclContext,
              const async_handler &AsyncHandler)
       : m_Context(SyclContext), m_AsyncHandler(AsyncHandler),
-        m_CommandQueue(CLQueue), m_OpenCLInterop(true), m_HostQueue(false) {
+        m_OpenCLInterop(true), m_HostQueue(false) {
```
[the remark this answers is not in the document]
Comment (smaslov-intel) [reply sycl/include/CL/sycl/detail/queue_impl.hpp]: Members are initialized in order of declaration in the class definition, not in the order of things in initializers list.

File: sycl/source/detail/helpers.cpp [hunk 4]
```diff
@@ -7,7 +7,6 @@
 //===----------------------------------------------------------------------===//
 
 #include <CL/sycl/detail/helpers.hpp>
-
```
[the remark this answers is not in the document]
Comment (smaslov-intel) [reply sycl/source/detail/helpers.cpp]: OK, thanks

File: sycl/doc/SYCL_environment_variables.md [hunk 5]
```diff
@@ -0,0 +1,11 @@
+# Overview
+
+This file describes environment variables that are having effect on SYCL compiler and run-time.
+
+# Controlling SYCL RT
+
+SYCL_PI_TRACE | If set forces tracing of PI calls to stdout.
+SYCL_BE={PI_OPENCL,PI_OTHER} | When SYCL RT is buils with PI this controls which plugin to use.
```
[the remark this answers is not in the document]
Comment (smaslov-intel) [reply sycl/doc/SYCL_environment_variables.md]: OK, is this this syntax: https://www.markdownguide.org/extended-syntax/#tables?

File: sycl/include/CL/sycl/detail/queue_impl.hpp (hunk 3 above)
[the remark this answers is not in the document]
Comment (bader) [reply sycl/include/CL/sycl/detail/queue_impl.hpp]: > Members are initialized in order of declaration in the class definition, not in the order of things in initializers list.

Right. I have exactly this warning in mind.

File: sycl/include/CL/sycl/detail/queue_impl.hpp [hunk 6]
```diff
@@ -132,18 +135,20 @@ class queue_impl {
...
         CL_QUEUE_PROPERTIES, CreationFlags, 0};
-    Queue = clCreateCommandQueueWithProperties(
-        ClContext, m_Device.get(), CreationFlagProperties,
-        &Error);
+    PI_CALL((Queue = RT::piQueueCreate(
+        Context, Device, CreationFlagProperties, &Error), Error));
 #else
-    Queue = clCreateCommandQueue(ClContext, m_Device.get(),
-                                          CreationFlags, &Error);
+    // TODO: do we really need this old interface into PI and here?
```
Comment (bader) [inline sycl/include/CL/sycl/detail/queue_impl.hpp]: No. This should be handled by OpenCL plugin. The plug-in should know where to lower `piQueueCreate` to `clCreateCommandQueueWithProperties` or to `clCreateCommandQueue`.

**Pull request merged**